presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2031,35 +2042,56 @@ public void run()
 
     private static class CreateTableOperation
     {
-        private final Table table;
+        private final Table newTable;
         private final PrincipalPrivileges privileges;
-        private boolean done;
+        private boolean tableCreated;
+        private final boolean ignoreExisting;
 
-        public CreateTableOperation(Table table, PrincipalPrivileges privileges)
+        public CreateTableOperation(Table newTable, PrincipalPrivileges privileges, boolean ignoreExisting)
         {
-            requireNonNull(table, "table is null");
-            checkArgument(getPrestoQueryId(table).isPresent());
-            this.table = table;
+            requireNonNull(newTable, "newTable is null");
+            checkArgument(getPrestoQueryId(newTable).isPresent());
+            this.newTable = newTable;
             this.privileges = requireNonNull(privileges, "privileges is null");
+            this.ignoreExisting = ignoreExisting;
         }
 
         public String getDescription()
         {
-            return format("add table %s.%s", table.getDatabaseName(), table.getTableName());
+            return format("add table %s.%s", newTable.getDatabaseName(), newTable.getTableName());
         }
 
         public void run(ExtendedHiveMetastore metastore)
         {
+            boolean done = false;
             try {
-                metastore.createTable(table, privileges);
+                metastore.createTable(newTable, privileges);
                 done = true;
             }
             catch (RuntimeException e) {
                 try {
-                    Optional<Table> remoteTable = metastore.getTable(table.getDatabaseName(), table.getTableName());
-                    // getPrestoQueryId(partition) is guaranteed to be non-empty. It is asserted in the constructor.
-                    if (remoteTable.isPresent() && getPrestoQueryId(remoteTable.get()).equals(getPrestoQueryId(table))) {
-                        done = true;
+                    Optional<Table> existingTable = metastore.getTable(newTable.getDatabaseName(), newTable.getTableName());
```

Comment:
See if you can write this like
```java
metastore.getTable(...).ifPresent(table -> {
    ...
}

Reply:
I wrote it initially like that, but based on the first set of comments I got I switched to this version.